### DISCUSSION TITLE
Images keep downloading after re-render

### DISCUSSION THREAD

**Discussion by Artemooon** [igordanchenko/react-photo-album#153] [Q&A] [answered]
How do you handle the situation when we load still the old images and the new ones after rerender the component?
It can be reproduced via changing the sorting, you can see in the networks that. 
When the images is heavy it takes a while 
_abort signal is not suitable in this case, because the request is done and images start downloading after the request_

**Comment by igordanchenko** [comment, marked as answer]
I'm not sure you can abort network requests in this case. Once the DOM gets updated with new images, it's up to the browser to cancel the mid-flight network requests (and it looks like most popular browsers don't do that). In this case, you can minimize network bandwidth waste by providing multiple images of various resolutions in the `srcSet` prop so that your browser can pick the most appropriate image resolution for your current viewport size.
